### DISCUSSION TITLE
Feature deprecation strategy

### DISCUSSION THREAD

**Discussion by joshbruce** [markedjs/marked#1114] [General]
**Marked version:** 0.3.17

**Proposal type:** other

## What pain point are you perceiving?

- We don't know how many users we truly have. 
- We don't know how many of our dependent projects are active or stale.
- We don't know how many of those dependents use a specific version of marked, or the latest "stable" (but broken) build.

## What solution are you suggesting?

- [ ] Devise as many ways possible to deprecate features while maintaining backward compatibility
- [ ] Devise and execute ways to discover answers to get the data we need while not negatively impacting the lives of the maintainers
- [ ] ...??

**Comment by UziTech** [comment]
I think the best way to deprecate features is to make marked like a core library and make plugins that provide the feature.

e.g. deprecate pedantic but build another package that updates the parser to provide pedantic for any customer that needs it.

**Comment by Martii** [comment]
> We don't know how many users we truly have.

This is a little abstract but falls with the realm of npmjs.com public stats e.g. https://npm-stat.com/charts.html?package=marked 

Security and privacy are paramount for a bunch. Even our site has engineered our analytics to be browser opted out. Putting in a privacy breach of trust could cause some very negative feedback if that is what is being considered.
